src/main/java/org/ecocompass/core/PathFinder/Query.java

```diff
@@ -104,6 +106,11 @@ private static void addPathModeRoute(List<TransitRoute> route, int lastIndex,
                                      RecommendationPath recommendation, String mode) {
         PathWithMode path = new PathWithMode();
         path.setMode(mode);
+        path.setStartStopName(route.get(lastIndex).getFoundSolution().getPossibleSolution().getStartNode().getName());
+        path.setEndStopName(route.get(lastIndex).getFoundSolution().getPossibleSolution().getEndNode().getName());
+        path.setModeNumber(route.get(lastIndex).getFoundSolution().getModeNumber());
+        path.setRouteNumber(route.get(lastIndex).getFoundSolution().getRoute());
+        path.setTimeStamp(route.get(lastIndex).getFoundSolution().getWaitTime().get(0));
```

Comment:
I am returning the most early available which is the next bus or luas available. hence the 0th one